### PR TITLE
Adding True money payment method

### PR DESCRIPTION
**Pull request by mayurkathale**
#### 1. Objective

Enable True Money Payment support.
More info about TrueMoney payment available [here](https://www.omise.co/truemoney)

#### 2. Description of change

Added Virtual and PHP classes to support True Money payment.
Added `PHTML` UI rendering files.

UI is very simple, and flow from a user perspective is the same as in the case of other payment method.
Added payment info in checkout page sidebar to show Phone number to be used for transaction.

1. Enable payment method in backend
Screen Shot 2020-01-14 at 12 03 13

2. Payment info section in checkout page
Screen Shot 2020-01-14 at 12 09 32

Screen Shot 2020-01-14 at 12 10 11

### 🔧 Environments:
Platform version: Magento CE 1.9
Omise plugin version: Omise-Magento 2.1.
PHP version: 5.6.
### ✏️ Details:
Turn on a True Money in the admin panel, check if payment is available on the checkout page.

Turn off a True Money method on the admin panel, the option should not be listed on the checkout page anymore.

Make an order and successful payment with the True Money method.

Verify if the proper amount and payment method are displayed in Omise Dashboard.

#### 4. Impact of the change

All Payment Method available for Thai merchants should work normally Magento 1

#### 5. Priority of change

Normal

#### 6. Additional Notes

None

### CODE REVIEW

File: src/app/design/frontend/base/default/template/payment/form/omise/omiseoffsitetruemoney.phtml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php $_code = $this->getMethodCode(); ?>
+
+<ul id="payment_form_<?php echo $_code; ?>" class="form-list" style="display:none;">
+    <!-- True money phone number form-->
+    <li>
+        <label for="<?php echo $_code; ?>_phone_number" class="required">
+            <em>*</em><?php echo $this->__('Phone Number'); ?>
+        </label>
+        <div class="input-box">
+            <input id="<?php echo $_code; ?>_phone_number" type="text"
```
Comment (jonrandy) [inline src/app/design/frontend/base/default/template/payment/form/omise/omiseoffsitetruemoney.phtml]: If the user is logged in and has a phone number in their default address, we should probably automatically have it filled in to the phone number field. I believe Jacek did something similar on Magento 2

Comment (mayurkathale) [reply src/app/design/frontend/base/default/template/payment/form/omise/omiseoffsitetruemoney.phtml]: @jonrandy it's done.

File: src/app/code/community/Omise/Gateway/Block/Form/Offsitetruemoney.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            array('block' => $this)
+        );
+        return parent::_toHtml();
+    }
+
+    /**
+     * Returns billing phone number from current quote
+     * @return string
+     */
+    public function getOmiseTruePhoneNumber() {
```
Comment (jonrandy) [inline src/app/code/community/Omise/Gateway/Block/Form/Offsitetruemoney.php]: I personally prefer the style of braces you've used (doesn't waste space), but apparently we have to use the other one (like the other functions)

File: src/app/code/community/Omise/Gateway/Block/Info/Truemoney.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    /**
+     * Prepare true money related payment info
+     *
+     * @param Varien_Object|array $transport
+     * @return Varien_Object
+     */
+    protected function _prepareSpecificInformation($transport = null)
+    {
+        if (null !== $this->_paymentSpecificInformation) {
```
Comment (jonrandy) [inline src/app/code/community/Omise/Gateway/Block/Info/Truemoney.php]: Greetings Master Yoda!

File: src/app/code/community/Omise/Gateway/etc/system.xml [hunk 4]
```diff
@@ -195,6 +195,27 @@
                                 </active>
                             </fields>
                         </omise_offsite_installment>
+
+                        <omise_offsite_truemoney type="group">
+                            <label>True Money</label>
+                            <comment>Enables customers to pay using True Money method.</comment>
```
Comment (jonrandy) [inline src/app/code/community/Omise/Gateway/etc/system.xml]: Can remove 'method' here

File: src/app/code/community/Omise/Gateway/Block/Info/Truemoney.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Prepare true money related payment info
+     *
+     * @param Varien_Object|array $transport
+     * @return Varien_Object
+     */
+    protected function _prepareSpecificInformation($transport = null)
+    {
+        if (null !== $this->_paymentSpecificInformation)
+        {
```
Comment (jonrandy) [inline src/app/code/community/Omise/Gateway/Block/Info/Truemoney.php]: That brace style doesn't apply to if statements apparently - just functions and classes

Comment (mayurkathale) [reply src/app/code/community/Omise/Gateway/Block/Info/Truemoney.php]: Reverted.

**Pull request merged**